flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java

```diff
@@ -617,12 +617,14 @@ public void testFilterNotPushDownLike() {
     explainNoPushDown = getTableEnv().explainSql(sqlNoPushDown);
     Assert.assertFalse("Explain should not contain FilterPushDown",
         explainNoPushDown.contains(expectedFilterPushDownExplain));
+
     sqlNoPushDown = "SELECT * FROM  " + TABLE_NAME + "  WHERE data LIKE '%%' ";
     resultLike = sql(sqlNoPushDown);
-    Assert.assertEquals("Should have 2 records", 2, resultLike.size());
+    Assert.assertEquals("Should have 3 records", 3, resultLike.size());
```

Comment:
Why do we need to change here ?  in the new flink 1.12.1 release,  the `null` data will be hit the `LIKE '%%'` ?

Reply:
yeah. there are a couple of small behavior changes in 1.12. I also confirmed with zhangjun0x01
https://github.com/apache/iceberg/commit/98d9f6ad511b47b554ca14ae5d7fa8931161b3bf

Reply:
I tested it again. In hive 2.3.7, the sql `select * from mytest where data like '%' ` , data is null will not match. In flink 1.12, it will return the value of data is null. I have not found the related flink jira for this